toolbox/include/base/ToolboxSingleton.h

```diff
@@ -0,0 +1,142 @@
+    /// name.
+    /// @see wbt::RobotInterface
+    std::unordered_map<std::string, std::shared_ptr<wbt::RobotInterface>> m_interfaces;
+
+public:
+
+    // CONSTRUCTOR / DESTRUCTOR
+    // ========================
+
+    ToolboxSingleton();
+    ToolboxSingleton(const ToolboxSingleton&);
+    ~ToolboxSingleton();
+    ToolboxSingleton& operator=(const ToolboxSingleton&);
+
+    // UTILITIES
+    // =========
+
+    /**
+     * Check if a Configuration labelled by confKey exists and is a
+     * valid object.
+     *
+     * @param  confKey The key describing the configuration (name of the Simulink block)
+     * @return         True if the configuration is valid
+     */
+    bool isKeyValid(const std::string& confKey);
+
+    /**
+     * Returns the DoFs associated with the configuration labelled by confKey
+     *
+     * @param  confKey The key describing the configuration (name of the Simulink block)
+     * @return         The number of degrees of freedom. It returns -1 when failing.
+     */
+    int numberOfDoFs(const std::string& confKey);
+
+    // GET METHODS
+    // ===========
+
+    /**
+     * Returns the singleton instance to this object.
+     *
+     * @note The singleton stays in memory until the entire matlab is closed. The WBToolbox
+     *       is designed in such a way that assures to reach a clean state when all blocks
+     *       have been terminated.
+     *
+     * @return the singleton instance
+     */
+    static wbt::ToolboxSingleton& sharedInstance();
+
+    // TODO: check if doxygen ref work
+
+    /**
+     * Returns the Configuration object labelled by confKey.
+     * This object is contained into the wbt::RobotInterface object.
+     *
+     * @see ToolboxSingleton::getRobotInterface
+     * @param  confKey The key describing the configuration (name of the Simulink block)
+     * @return         A constant reference to the Configuration object
+     */
+    const Configuration& getConfiguration(const std::string& confKey);
+
+    /**
+    * Returns a \c shared_ptr to the RobotInterface object containing the configuration
+    * labelled by confKey and all the objects used to gather and set robot data
+    *
+    * @param confKey The key describing the configuration (name of the Simulink block)
+    * @return        A \c shared_ptr to the RobotInterface of the requested configuration
+    */
+    const std::shared_ptr<RobotInterface> getRobotInterface(const std::string& confKey);
+
+    /**
+     * Returns a \c shared_ptr to the KinDynComputations object used to perform calculation
+     * on the provided model
+     *
+     * @param confKey The key describing the configuration (name of the Simulink block)
+     * @return        A \c shared_ptr to the iDynTree::KinDynComputations of the requested
+     *                configuration
+     */
+    const std::shared_ptr<iDynTree::KinDynComputations> model(const std::string& confKey);
```

Comment:
To be consistent with the previous two methods I would rename this into `getModel`

Reply:
This was an inheritance of the previous version, changed the name. This method might even be removed, it's never called. Right now the best way to gather the model is asking it directly from the parent `WBBlock` using `getRobotInterface()->getKinDynComputations()`. This allows the block not to know the existence of the `confKey` variable.